### PR TITLE
Templates: Renames template directories

### LINKED ISSUE #7698: Restructure story templates
## Feature Description

Restructure templates package to facilitate the addition of more templates in future.

## Additional Context

At present, there are only 8 story templates however, going forward the number is going to increase with 54 new Story Templates being created. 

The static assets for these story templates are added directly to the vertical ( category ) directories. But going onward since we are going to have multiple stories under these verticals, we may need to re-arrange the existing static assets in a way that can scale with the increasing number of story templates.

Static Assets:
> https://github.com/google/web-stories-wp/tree/static-site/public/static/main/images/templates/

Stickers: 
> https://github.com/google/web-stories-wp/tree/main/packages/stickers/src

Raw Templates:
> https://github.com/google/web-stories-wp/tree/main/packages/templates/src/raw

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by divyarajmasani**
## Context

To accommodate the increasing number of templates, we may consider restructuring how the templates are loaded.

## Summary

- Renames vertical based directories for raw templates and stickers, to use title from the first page of the story template.
- Updates media asset paths to use new changes introduced in #7695 
- Updates templates package tests for new changes.

## Relevant Technical Choices

This changes also prompted to make changes to the way all templates were loaded, raw template json will now be placed under `packages/templates/src/raw/<template_name>` directory. `<template_name>` will be the title shown on the first page of the story template.

```
-- <template_name>
  | --- index.js
  | --- template.json
```

`index.js` is where the template object is exported from and has definition as shown below. 

```js

/**
 * Internal dependencies
 */
import { default as template } from './template';

export default {
  title: '',
  tags: [],
  colors: [],
  description:'',
  ...template,
  vertical: '',
};
```

## To-do

n/a

## User-facing changes

n/a

## Testing Instructions

- Navigate to 'Explore Templates' on admin side.
- Verify that all eight story templates are present and working as expected.

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #

### CODE REVIEW

File: packages/templates/src/raw/sleep/index.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ */
+import { __, _x } from '@web-stories-wp/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { default as template } from './template';
+
+export default {
+  title: _x('Wellbeing: Sleep', 'template name', 'web-stories'),
```
Comment (swissspidy) [inline packages/templates/src/raw/sleep/index.js]: Why not just `Sleep`?

The other templates don't have `<vertical>: <name>` titles

Comment (divyarajmasani) [reply packages/templates/src/raw/sleep/index.js]: Yes, you are right.

Just `Sleep` felt too generic, and to add a context I added the `Wellbeing`. But can remove it.

File: packages/stickers/src/index.js [hunk 2]
```diff
@@ -16,22 +16,22 @@
...
+import * as beautyStickers from './fresh-and-bright';
+import * as cookingStickers from './food-and-stuff';
+import * as doersGetMoreDoneStickers from './doers-get-more-done';
+import * as weeklyEntertainmentStickers from './weekly-entertainment';
+import * as fashionOnTheGoStickers from './fashion-on-the-go';
+import * as noDaysOffStickers from './no-days-off';
+import * as experienceThailandStickers from './experience-thailand';
+import * as wellbeingSleepStickers from './sleep';
 
 export default {
```
Comment (swissspidy) [inline packages/stickers/src/index.js:28]: Related question: how well does this scale once we have 50 templates (presumably all loaded from our CDN)?

cc @littlemilkstudio

**Comment by swissspidy** [comment]
@divyarajmasani Could you perhaps update `docs/external-template-creation.md` accordingly? Specifically the `Adding story JSON to the codebase as a new template` section.

**Review by swissspidy — APPROVED** [review]

File: packages/templates/src/raw/sleep/index.js (hunk 1 above)
Comment (swissspidy) [reply packages/templates/src/raw/sleep/index.js]: Let's remove it for consistency 👍 

`template name` should be enough context

**Comment by swissspidy** [comment]
@littlemilkstudio @choumx Added you as reviewers just as an FYI. We need to restructure the template directories to better accommodate all the new templates we'll be introducing.

Before, we just used the vertical as the template name, but now we have more unique template names.

File: packages/templates/src/raw/sleep/index.js (hunk 1 above)
Comment (divyarajmasani) [reply packages/templates/src/raw/sleep/index.js]: Updated the title.

**Comment by divyarajmasani** [comment]
> @divyarajmasani Could you perhaps update `docs/external-template-creation.md` accordingly? Specifically the `Adding story JSON to the codebase as a new template` section.

@swissspidy I've updated the `docs` for adding JSON and stickers.

**Comment by dreamofabear** [comment]
Did we not want to retain a directory level for template vertical (entertainment, food, etc.)?

**Comment by swissspidy** [comment]
> Did we not want to retain a directory level for template vertical (entertainment, food, etc.)?

We've considered it, but having this additional level didn't seem advantageous at this time. It seems appealing if all 50 templates would be in here, sure, but I am planning on moving all the templates to Firebase as soon as possible (see #7196), where there's no need for a hierarchical structure.

**Pull request merged**